build.sh

```diff
@@ -0,0 +1,130 @@
+export PATH=${HOME}/.local/bin:${PATH}
+python3 -m pip install --user -r lint-requirements.txt
+
+# If this script is run in a container and the user in the container doesn't
+# match the owner of the Git checkout, then Git objects; these config settings
+# prevent the complaints.
+GITTOP=$(git rev-parse --show-toplevel 2>&1 | head -n 1)
+if [[ ${GITTOP} =~ "fatal: unsafe repository ('/home/root/pbench'" ]] ; then
+	git config --global --add safe.directory /home/root/pbench
+	git config --global --add safe.directory /home/root/pbench/agent/stockpile
+	GITTOP=$(git rev-parse --show-toplevel)
+fi
+
+# Install the Dashboard dependencies, including the linter's dependencies and
+# the unit test dependencies.
+( cd dashboard && npm install )
+
+# Test for code style and lint
+black --check .
+flake8 .
+( cd dashboard && npx eslint "src/**" --max-warnings 0 )
+
+# Run unit tests
+tox                                     # Agent and Server unit tests and legacy tests
+( cd dashboard && CI=true npm test )    # Dashboard unit tests
+
+# Build Agent and Server RPMs (Server RPM includes the Dashboard) on COPR
+#
+# TODO:  We should set up or maintain a ./seqno file.
+( cd agent/rpm && make COPR_USER=${COPR_USER} COPR_CONFIG=${COPR_CONFIG} copr-ci )
+( cd server/rpm && make COPR_USER=${COPR_USER} COPR_CONFIG=${COPR_CONFIG} copr-ci )
+
+# The following steps don't work inside the CI container, so just declare
+# victory now.  (Building a container inside a container requires a
+# specially-configured and -invoked container and/or privilege, and we don't
+# have that arranged, yet.)
+exit 0
+
+################################################################################
+#  Future extensions
+################################################################################
+
+# Need to log into the registry to access the Server base image, etc.
+podman login -u="${PODMAN_USR}" -p="${PODMAN_PSW}" quay.io
+
+# Build the Agent container images and the Pbench-in-a-Can Server image
+( cd agent/containers/images && \
+  make COPR_USER=${COPR_USER} URL_PREFIX=${URL_PREFIX} TEST=ci \
+  )
+server/pbenchinacan/container-build.sh
+
+# TODO:  do we need/want to pre-load sample data into the canned Server?
+
+# Push the Agent and Server container images to the container registry.
+( cd agent/containers/images && \
+  make COPR_USER=${COPR_USER} URL_PREFIX=${URL_PREFIX} TEST=ci tag-ci push-ci \
+  )
+podman push quay.io/pbench/pbenchinacan-pbenchserver
+
+# FIXME:  We need a consistent set of container images for the Pbench Server,
+# PostgreSQL, and Elasticsearch, in terms of the data that they contain.  If
+# we're going to go with an "empty" Server, then we need to ensure that the
+# images on Quay.io for the PostgreSQL and Elasticsearch containers are empty,
+# as well, but we don't need to push them every time we rebuild the Server;
+# otherwise, we need to commit the seeded images for all three and push them up.
```

Comment:
Yeah; maybe start out with empty backend DBs. We can unpack a few of the legacy test tarballs (or just make a few new ones) and build a simple script to `PUT` them to the new server, let them be indexed, and then do some queries.

Reply:
I've already dealt with the problem of pre-loading data into the canned Server.  I just have to enshrine the process in code here.  Obviously it would be simpler to spin up an empty Server, as doing so would allow us to skip the pre-load steps, but the preloading isn't hard to do.  What we should do depends on what we want the functional testing to look like and at what point we want it to start from.

Given that the functional testing is presumably going to load data, my thinking was that there's probably no point (in this context) to pre-loading data....